hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -622,9 +623,10 @@ object DataSourceWriteOptions {
   /** @deprecated Use {@link PRECOMBINE_FIELD} and its methods instead */
   @Deprecated
   val PRECOMBINE_FIELD_OPT_KEY = HoodieWriteConfig.PRECOMBINE_FIELD_NAME.key()
-  /** @deprecated Use {@link PRECOMBINE_FIELD} and its methods instead */
+  /** @deprecated Use {@link PRECOMBINE_FIELD} and its methods instead.
+   *             This field has no default value since version 0.12.0, `ts` is for backward compatibility. */
   @Deprecated
-  val DEFAULT_PRECOMBINE_FIELD_OPT_VAL = PRECOMBINE_FIELD.defaultValue()
+  val DEFAULT_PRECOMBINE_FIELD_OPT_VAL = "ts"
```

Comment:
should we even kill the defaults defined here.

Reply:
It's there just for BWC. Will take it in a separate PR that removes all deprecated configs.